test/libsolidity/util/TestFileParser.cpp

```diff
@@ -169,6 +171,23 @@ vector<solidity::frontend::test::FunctionCall> TestFileParser::parseFunctionCall
 	return calls;
 }
 
+vector<string> TestFileParser::parseFunctionCallSideEffects()
+{
+	vector<string> result;
+	while (accept(Token::Tilde, false))
+	{
+		string effect = m_scanner.currentLiteral();
+		result.emplace_back(effect);
```

Comment:
I'm trying to understand this but I'm not sure why it works. `accept()` will return `true` when `m_scanner.currentToken() == Token::Tilde` and `m_scanner.currentLiteral()` will return `"~"` in that case, right? But it would mean that `result` is a vector of tildes and that does not seem to be the case. Why?

Reply:
The scanner's token `m_currentToken` and it's literal `m_currentLiteral` is set within `TestFileParser::Scanner::scanNextToken()`, e.g.:
```cpp
auto selectToken = [this](Token _token, std::string const& _literal = "") {
  advance();
  m_currentToken = _token;
  m_currentLiteral = _literal;
};
...
switch(current())
{
  ...
  case '~':
	advance();
	selectToken(Token::Tilde, readLine());
	break;
   ...
}
```
where `selectToken` will set the current token, but it will also set the corresponding literal. In the case of side-effects, we use the `readLine()` function to set the literal. (Previously that `readLine()` functions was named `scanEffectString()`). 

Thats why we will have `m_scanner.currentToken() == Token::Tilde`, where `m_scanner.currentLiteral()` will contain "the rest of that line" - excluding `~` and it's next character - we use `advance()` here - before actually reading the line by calling `readLine`.